<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yunqi Yang" />

<meta name="date" content="2019-11-19" />

<title>mashr_application</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mash-single-cell-rnaseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">mashr_application</h1>
<h4 class="author"><em>Yunqi Yang</em></h4>
<h4 class="date"><em>11/19/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-11-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mash-single-cell-rnaseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191120code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191120)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191120code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191120)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/top_snps.RData
</td>
<td style="text-align:left;">
data/top_snps.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabmashsinglecellrnaseqtree87a59f659b80373a553aedd3cc0de881f45c4e64targetblank87a59f6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/mash-single-cell-rnaseq/tree/87a59f659b80373a553aedd3cc0de881f45c4e64" target="_blank">87a59f6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabmashsinglecellrnaseqtree87a59f659b80373a553aedd3cc0de881f45c4e64targetblank87a59f6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/top_snps.RData
    Untracked:  top_snps.RData

Unstaged changes:
    Modified:   analysis/RNA_seq_process2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/87a59f659b80373a553aedd3cc0de881f45c4e64/analysis/mashr_application.Rmd" target="_blank">87a59f6</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2019-11-21
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”, verbose = TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Remarks:</p>
<ol style="list-style-type: decimal">
<li><p>Mashr re-estimate the effect size of SNPs, incorporating information across conditions.</p></li>
<li><p>Ways to access the fit. Log-likelihood and others?</p></li>
<li><p>Seems that result doesn’t make sense. Log-likelihood very large.</p></li>
</ol>
<pre class="r"><code>library(ashr)
library(mashr)</code></pre>
<div id="format-data-to-mashr-format" class="section level2">
<h2>format data to mashr format</h2>
<pre class="r"><code>load(&quot;/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/top_snps.RData&quot;)

dt_beta &lt;- top_snps[,c(&quot;beta_naive&quot;, &quot;beta_IF&quot;, &quot;beta_IFSL&quot;, &quot;beta_SL&quot;)]
dt_pval &lt;- top_snps[,c(&quot;p_nominal_naive&quot;, &quot;p_nominal_IF&quot;, &quot;p_nominal_IFSL&quot;, &quot;p_nominal_SL&quot;)]

dt_beta = as.matrix(dt_beta)
dt_pval = as.matrix(dt_pval)

head(dt_beta)</code></pre>
<pre><code>  beta_naive    beta_IF  beta_IFSL    beta_SL
1 -0.3370090 -0.4037950 -0.0941426 -0.2029850
2 -0.0743763 -0.0419751 -0.0325720 -0.0214099
3 -0.0786190 -0.0507808 -0.0591935 -0.0275128
4  0.0826863 -0.0162367 -0.0407589  0.0786256
5 -0.4299870 -0.3545930 -0.2523280 -0.2987360
6 -0.3760470 -0.0694328  0.0700316 -0.0902658</code></pre>
<pre class="r"><code>head(dt_pval)</code></pre>
<pre><code>  p_nominal_naive p_nominal_IF p_nominal_IFSL p_nominal_SL
1      0.00061327   0.00140049      0.5007650    0.0545107
2      0.00121529   0.08250130      0.1340060    0.3738330
3      0.00117232   0.04579800      0.0610855    0.5071570
4      0.00256488   0.68539900      0.5733360    0.0466930
5      0.03554660   0.03811650      0.0227163    0.0728410
6      0.01376920   0.37466200      0.2537470    0.5845360</code></pre>
<pre class="r"><code>str(dt_beta)</code></pre>
<pre><code> num [1:15678, 1:4] -0.337 -0.0744 -0.0786 0.0827 -0.43 ...
 - attr(*, &quot;dimnames&quot;)=List of 2
  ..$ : chr [1:15678] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
  ..$ : chr [1:4] &quot;beta_naive&quot; &quot;beta_IF&quot; &quot;beta_IFSL&quot; &quot;beta_SL&quot;</code></pre>
<pre class="r"><code>str(dt_pval)</code></pre>
<pre><code> num [1:15678, 1:4] 0.000613 0.001215 0.001172 0.002565 0.035547 ...
 - attr(*, &quot;dimnames&quot;)=List of 2
  ..$ : chr [1:15678] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
  ..$ : chr [1:4] &quot;p_nominal_naive&quot; &quot;p_nominal_IF&quot; &quot;p_nominal_IFSL&quot; &quot;p_nominal_SL&quot;</code></pre>
<pre class="r"><code>dt_mash = mash_set_data(dt_beta, Shat = NULL, pval = dt_pval)
head(dt_mash$Bhat)</code></pre>
<pre><code>  beta_naive    beta_IF  beta_IFSL    beta_SL
1 -0.3370090 -0.4037950 -0.0941426 -0.2029850
2 -0.0743763 -0.0419751 -0.0325720 -0.0214099
3 -0.0786190 -0.0507808 -0.0591935 -0.0275128
4  0.0826863 -0.0162367 -0.0407589  0.0786256
5 -0.4299870 -0.3545930 -0.2523280 -0.2987360
6 -0.3760470 -0.0694328  0.0700316 -0.0902658</code></pre>
<pre class="r"><code>head(dt_mash$Shat)</code></pre>
<pre><code>  beta_naive    beta_IF  beta_IFSL    beta_SL
1 0.09837735 0.12640121 0.13982546 0.10556982
2 0.02298923 0.02417428 0.02173655 0.02407456
3 0.02422377 0.02542536 0.03160567 0.04148048
4 0.02741970 0.04008005 0.07237725 0.03952874
5 0.20455348 0.17100359 0.11075895 0.16653569
6 0.15266426 0.07820994 0.06136150 0.16508883</code></pre>
<pre class="r"><code># Step2: set up covariance matrix 
U.c = cov_canonical(dt_mash)  
print(names(U.c))</code></pre>
<pre><code>[1] &quot;identity&quot;      &quot;beta_naive&quot;    &quot;beta_IF&quot;       &quot;beta_IFSL&quot;    
[5] &quot;beta_SL&quot;       &quot;equal_effects&quot; &quot;simple_het_1&quot;  &quot;simple_het_2&quot; 
[9] &quot;simple_het_3&quot; </code></pre>
<pre class="r"><code># Step3: fit model 
m.c = mash(dt_mash, U.c)</code></pre>
<pre><code> - Computing 15678 x 253 likelihood matrix.
 - Likelihood calculations took 2.62 seconds.
 - Fitting model with 253 mixture components.
 - Model fitting took 15.28 seconds.
 - Computing posterior matrices.
 - Computation allocated took 13.21 seconds.</code></pre>
<pre class="r"><code># Step4: extract posterior summarize 

head(get_lfsr(m.c))   # local false sign rates</code></pre>
<pre><code>    beta_naive     beta_IF   beta_IFSL     beta_SL
1 1.461406e-05 0.001824021 0.004385345 0.002141932
2 1.876247e-04 0.099080350 0.099233110 0.099949419
3 3.520635e-05 0.016831278 0.016881393 0.018060472
4 2.629536e-03 0.495295005 0.494371262 0.468454622
5 4.657397e-04 0.001811972 0.001688937 0.001921296
6 7.303887e-02 0.855488753 0.873244307 0.858295958</code></pre>
<pre class="r"><code>head(get_pm(m.c))</code></pre>
<pre><code>   beta_naive      beta_IF    beta_IFSL      beta_SL
1 -0.24998712 -0.250345427 -0.243688943 -0.245792041
2 -0.04582012 -0.038363102 -0.038302041 -0.038231226
3 -0.05954638 -0.058071103 -0.058122522 -0.057942265
4  0.06609349  0.024188691  0.024158574  0.027910831
5 -0.26644629 -0.265655730 -0.264519032 -0.265028508
6 -0.19970374 -0.005753663 -0.002717302 -0.005782514</code></pre>
<pre class="r"><code>head(get_psd(m.c))</code></pre>
<pre><code>  beta_naive    beta_IF  beta_IFSL    beta_SL
1 0.05799366 0.06082965 0.06221880 0.05897826
2 0.01624149 0.01686986 0.01685286 0.01691490
3 0.01488694 0.01620381 0.01625526 0.01650750
4 0.02791045 0.02987130 0.03139002 0.03048489
5 0.07604640 0.07563894 0.07428845 0.07540395
6 0.13864721 0.02489209 0.02284952 0.02834736</code></pre>
<pre class="r"><code>head(get_significant_results(m.c))</code></pre>
<pre><code> 661 2075 2694 3191 3304 3305 
 661 2068 2687 3184 3297 3298 </code></pre>
<pre class="r"><code>print(length(get_significant_results(m.c)))</code></pre>
<pre><code>[1] 15450</code></pre>
</div>
<div id="get-sharing" class="section level2">
<h2>get sharing</h2>
<pre class="r"><code>print(get_pairwise_sharing(m.c, factor=0))</code></pre>
<pre><code>           beta_naive   beta_IF beta_IFSL   beta_SL
beta_naive  1.0000000 0.9922977 0.9803883 0.9858900
beta_IF     0.9922977 1.0000000 0.9977311 0.9984487
beta_IFSL   0.9803883 0.9977311 1.0000000 0.9989194
beta_SL     0.9858900 0.9984487 0.9989194 1.0000000</code></pre>
<pre class="r"><code>print(get_loglik(m.c))</code></pre>
<pre><code>[1] 39976.75</code></pre>
</div>
<div id="estimate-mixture-proportion" class="section level2">
<h2>estimate mixture proportion</h2>
<pre class="r"><code>print(get_estimated_pi(m.c))</code></pre>
<pre><code>         null      identity    beta_naive       beta_IF     beta_IFSL 
 0.0006367433  0.0000000000  0.2126963603  0.0000000000  0.0000000000 
      beta_SL equal_effects  simple_het_1  simple_het_2  simple_het_3 
 0.0000000000  0.7166936937  0.0000000000  0.0000000000  0.0699732027 </code></pre>
<pre class="r"><code>barplot(get_estimated_pi(m.c),las = 2)</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.15

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mashr_0.2.21.0641 ashr_2.2-38      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2        plyr_1.8.4        compiler_3.5.1   
 [4] later_1.0.0       git2r_0.26.1      highr_0.7        
 [7] workflowr_1.5.0   iterators_1.0.12  tools_3.5.1      
[10] digest_0.6.18     evaluate_0.13     lattice_0.20-38  
[13] rlang_0.4.0       Matrix_1.2-15     foreach_1.4.7    
[16] yaml_2.2.0        parallel_3.5.1    mvtnorm_1.0-11   
[19] xfun_0.4          stringr_1.4.0     knitr_1.21       
[22] fs_1.3.1          rprojroot_1.3-2   grid_3.5.1       
[25] glue_1.3.0        R6_2.4.0          rmarkdown_1.11   
[28] mixsqp_0.1-97     rmeta_3.0         magrittr_1.5     
[31] whisker_0.3-2     backports_1.1.3   promises_1.1.0   
[34] codetools_0.2-16  htmltools_0.4.0   MASS_7.3-51.1    
[37] assertthat_0.2.1  abind_1.4-5       httpuv_1.5.2     
[40] stringi_1.3.1     doParallel_1.0.15 pscl_1.5.2       
[43] truncnorm_1.0-8   SQUAREM_2017.10-1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
