<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yunqi Yang" />


<title>mashr_application1</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mash-single-cell-rnaseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">mashr_application1</h1>
<h4 class="author"><em>Yunqi Yang</em></h4>
<h4 class="date"><em>12/8/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mash-single-cell-rnaseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191120code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191120)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191120code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191120)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/top_snps.RData
</td>
<td style="text-align:left;">
data/top_snps.RData
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/random_snps.RData
</td>
<td style="text-align:left;">
data/random_snps.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabmashsinglecellrnaseqtreebd319b837969339c2586c19bcd0536ccea28bf70targetblankbd319b8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/mash-single-cell-rnaseq/tree/bd319b837969339c2586c19bcd0536ccea28bf70" target="_blank">bd319b8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabmashsinglecellrnaseqtreebd319b837969339c2586c19bcd0536ccea28bf70targetblankbd319b8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  GSE71220_series_matrix.txt.gz
    Untracked:  analysis/random_snps.RData
    Untracked:  data/dt_all.RData
    Untracked:  data/random_snps.RData
    Untracked:  data/top_snps.RData
    Untracked:  top_snps.RData

Unstaged changes:
    Modified:   analysis/RNA_seq_process2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/bd319b837969339c2586c19bcd0536ccea28bf70/analysis/mashr_application.Rmd" target="_blank">bd319b8</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-14
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/mash-single-cell-rnaseq/36610918f820678e90add4d5229b0ca97b4c278e/docs/mashr_application.html" target="_blank">3661091</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/c2c7daaf7e2119067ff26264774c1e4b1612e95a/analysis/mashr_application.Rmd" target="_blank">c2c7daa</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-09
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/mash-single-cell-rnaseq/dcbc07a21407feaa16a9fa7fbe7e943647660a13/docs/mashr_application.html" target="_blank">dcbc07a</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/mash-single-cell-rnaseq/ed57c109db205a5894462ef3b09e3204b8201e35/docs/mashr_application.html" target="_blank">ed57c10</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/07625ebff67bbeaf65eb1804fb28340e5ac743c3/analysis/mashr_application.Rmd" target="_blank">07625eb</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/mash-single-cell-rnaseq/f74b3e41394c08da7f8898fa52ff722924b90be0/docs/mashr_application.html" target="_blank">f74b3e4</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/892e43b074077fbb5670c7640359017a0aba5800/analysis/mashr_application.Rmd" target="_blank">892e43b</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/mash-single-cell-rnaseq/59bbc5e2bdc3acd37548ad94a5a5fa19c3dc8aea/docs/mashr_application.html" target="_blank">59bbc5e</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2019-11-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/87a59f659b80373a553aedd3cc0de881f45c4e64/analysis/mashr_application.Rmd" target="_blank">87a59f6</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2019-11-21
</td>
<td>
wflow_publish(“analysis/mashr_application.Rmd”, verbose = TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Remarks:</p>
<ol style="list-style-type: decimal">
<li><p>Used data from “Shared genetic effects on chromatin and gene expression indicate a role for enhancer priming in immune response” <a href="https://doi.org/10.1038/s41588-018-0046-7" class="uri">https://doi.org/10.1038/s41588-018-0046-7</a>. eQTLs from 4 conditions downloaded from <a href="https://zenodo.org/record/1158560#.Xgjl5BdKhp8" class="uri">https://zenodo.org/record/1158560#.Xgjl5BdKhp8</a>.</p></li>
<li><p>For selecting strong signals: we first filter with the criteria p-value &lt;0.05. Then, take the top signals (the most significant SNP for each gene) in naive condition, and filter gene-SNP pairs in other conditions based on top signals in naive condition. The resulting “top_snps.RData” contains 15678 significant Gene-SNP pairs in 4 conditions.</p></li>
<li><p>The 4 conditions are: naive/ IF/ IF+SL/ SL. (IF, SL are two drug treatments)</p></li>
<li><ol style="list-style-type: decimal">
<li>Use strong signals to learn data-driven covariance patterns. (2) Fit models on the random subset of data, to learn weights for data-driven covariance matrix + canonical covariance matrix.</li>
</ol></li>
<li><p>Ways to assess the model fit.</p></li>
<li><p>Compared the log-likelihood with/without ED step: extremely similar.</p></li>
</ol>
<pre class="r"><code>library(ashr)
library(mashr)</code></pre>
<div id="format-data-to-mashr-format" class="section level2">
<h2>Format data to mashr format</h2>
<pre class="r"><code>load(&quot;/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/top_snps.RData&quot;)
load(&quot;/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/random_snps.RData&quot;)

dt_beta_top &lt;- top_snps[,c(&quot;beta_naive&quot;, &quot;beta_IF&quot;, &quot;beta_IFSL&quot;, &quot;beta_SL&quot;)]
dt_pval_top &lt;- top_snps[,c(&quot;p_nominal_naive&quot;, &quot;p_nominal_IF&quot;, &quot;p_nominal_IFSL&quot;, &quot;p_nominal_SL&quot;)]


dt_beta_random &lt;- random_snps[,c(&quot;beta_naive&quot;, &quot;beta_IF&quot;, &quot;beta_IFSL&quot;, &quot;beta_SL&quot;)]
dt_pval_random &lt;- random_snps[,c(&quot;p_nominal_naive&quot;, &quot;p_nominal_IF&quot;, &quot;p_nominal_IFSL&quot;, &quot;p_nominal_SL&quot;)]

dt_beta_top = as.matrix(dt_beta_top)
dt_pval_top = as.matrix(dt_pval_top)

dt_beta_random = as.matrix(dt_beta_random)
dt_pval_random = as.matrix(dt_pval_random)</code></pre>
</div>
<div id="correlation-structure" class="section level2">
<h2>Correlation structure</h2>
<pre class="r"><code>data.temp = mash_set_data(dt_beta_random, Shat = NULL, pval = dt_pval_random)
Vhat = estimate_null_correlation_simple(data.temp)
rm(data.temp)</code></pre>
<pre class="r"><code>data.strong = mash_set_data(dt_beta_top, Shat = NULL, pval = dt_pval_top, V=Vhat)
data.random = mash_set_data(dt_beta_random, Shat = NULL, pval = dt_pval_random, V=Vhat)</code></pre>
</div>
<div id="data-driven-covariance" class="section level2">
<h2>Data driven covariance</h2>
<pre class="r"><code>U.pca = cov_pca(data.strong,4)
U.ed = cov_ed(data.strong, U.pca)</code></pre>
</div>
<div id="fit-mashr-model-with-both-canonical-data-driven-covariance-structure" class="section level2">
<h2>Fit mashr model with both canonical &amp; data-driven covariance structure</h2>
<pre class="r"><code>U.c = cov_canonical(data.random)
m = mash(data.random, Ulist = c(U.ed,U.c))</code></pre>
<pre><code> - Computing 40000 x 393 likelihood matrix.
 - Likelihood calculations took 10.09 seconds.
 - Fitting model with 393 mixture components.
 - Model fitting took 83.44 seconds.
 - Computing posterior matrices.
 - Computation allocated took 139.36 seconds.</code></pre>
<pre class="r"><code>## Use learned weights to gain posterior estimation for strong signals
m2 = mash(data.strong, g=get_fitted_g(m), fixg=TRUE)</code></pre>
<pre><code> - Computing 15678 x 393 likelihood matrix.
 - Likelihood calculations took 3.83 seconds.
 - Computing posterior matrices.
 - Computation allocated took 19.45 seconds.</code></pre>
</div>
<div id="estimate-data-driven-covariance" class="section level2">
<h2>Estimate data driven covariance</h2>
<pre class="r"><code>U.pca = cov_pca(data.strong,4)
U.ed = cov_ed(data.strong, U.pca)</code></pre>
</div>
<div id="fit-mashr-model-with-both-canonical-data-driven-covariance-structure-1" class="section level2">
<h2>Fit mashr model with both canonical &amp; data-driven covariance structure</h2>
<pre class="r"><code>U.c = cov_canonical(data.random)
m = mash(data.random, Ulist = c(U.ed,U.c))</code></pre>
<pre><code> - Computing 40000 x 393 likelihood matrix.
 - Likelihood calculations took 9.21 seconds.
 - Fitting model with 393 mixture components.
 - Model fitting took 85.87 seconds.
 - Computing posterior matrices.
 - Computation allocated took 150.04 seconds.</code></pre>
<pre class="r"><code>## Use learned weights to gain posterior estimation for strong signals
m2 = mash(data.strong, g=get_fitted_g(m), fixg=TRUE)</code></pre>
<pre><code> - Computing 15678 x 393 likelihood matrix.
 - Likelihood calculations took 3.71 seconds.
 - Computing posterior matrices.
 - Computation allocated took 20.70 seconds.</code></pre>
<pre class="r"><code>print(get_loglik(m))</code></pre>
<pre><code>[1] 205233</code></pre>
<pre class="r"><code>print(get_estimated_pi(m))</code></pre>
<pre><code>         null      ED_PCA_1      ED_PCA_2      ED_PCA_3      ED_PCA_4 
 3.865163e-01  5.281307e-03  1.511466e-04  0.000000e+00  3.692821e-02 
      ED_tPCA      identity    beta_naive       beta_IF     beta_IFSL 
 2.683872e-04  3.302612e-05  0.000000e+00  0.000000e+00  0.000000e+00 
      beta_SL equal_effects  simple_het_1  simple_het_2  simple_het_3 
 0.000000e+00  5.530311e-01  0.000000e+00  2.458072e-04  1.754470e-02 </code></pre>
<pre class="r"><code>barplot(get_estimated_pi(m),las = 2)</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/f74b3e41394c08da7f8898fa52ff722924b90be0/docs/figure/mashr_application.Rmd/unnamed-chunk-9-1.png" target="_blank">f74b3e4</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mash_plot_meta(m,get_significant_results(m)[1])</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/mash-single-cell-rnaseq/blob/f74b3e41394c08da7f8898fa52ff722924b90be0/docs/figure/mashr_application.Rmd/unnamed-chunk-10-1.png" target="_blank">f74b3e4</a>
</td>
<td>
Nicholeyang0215
</td>
<td>
2020-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="effect-size-comparison" class="section level2">
<h2>Effect Size Comparison</h2>
<div id="posterior-mean-vs.-original-effect" class="section level3">
<h3>Posterior Mean VS. Original Effect</h3>
<p>Conditions 1-4 are: naive/ IF/ IF+SL/ SL. (IF, SL are two drug treatments)</p>
<div id="random-signal-posterior-comparison" class="section level4">
<h4>random signal posterior comparison</h4>
<pre class="r"><code>par(mfrow = c(2,2))

for (i in c(1:4)){
  
  plot(data.random$Bhat[,i], get_pm(m)[,i], pch = 20, ylab = &quot;Posterior Mean&quot;, xlab = &quot;Original Effect&quot;, main = paste(&#39;Condition_&#39;,i, sep = &quot;&quot;))
  abline(coef = c(0,1), col = &quot;red&quot;)
  
}</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="scaled-by-standard-deviation-posterior-mean-vs.-original-effect" class="section level4">
<h4>Scaled by standard deviation: Posterior Mean VS. Original Effect</h4>
<pre class="r"><code>par(mfrow = c(2,2))
for ( i in c(1:4)){
  
  plot(data.random$Bhat[,i]/data.random$Shat[,i], get_pm(m)[,i]/get_psd(m)[,i], pch = 20, ylab = &quot;Posterior Mean&quot;, xlab = &quot;Original Effect&quot;, main = paste(&#39;Condition_&#39;,i, sep = &quot;&quot;))
  abline(coef = c(0,1), col = &quot;red&quot;)
  
}</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="strong-signal-posterior-comparison" class="section level4">
<h4>strong signal posterior comparison</h4>
<pre class="r"><code>par(mfrow = c(2,2))

for (i in c(1:4)){
  
  plot(data.strong$Bhat[,i], get_pm(m2)[,i], pch = 20, ylab = &quot;Posterior Mean&quot;, xlab = &quot;Original Effect&quot;, main = paste(&#39;Condition_&#39;,i, sep = &quot;&quot;))
  abline(coef = c(0,1), col = &quot;red&quot;)
  
}</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="scaled-by-standard-deviation-posterior-mean-vs.-original-effect-1" class="section level4">
<h4>Scaled by standard deviation: Posterior Mean VS. Original Effect</h4>
<pre class="r"><code>par(mfrow = c(2,2))
for ( i in c(1:4)){
  
  plot(data.strong$Bhat[,i]/data.strong$Shat[,i], get_pm(m2)[,i]/get_psd(m2)[,i], pch = 20, ylab = &quot;Posterior Mean&quot;, xlab = &quot;Original Effect&quot;, main = paste(&#39;Condition_&#39;,i, sep = &quot;&quot;))
  abline(coef = c(0,1), col = &quot;red&quot;)
  
}</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="run-mashr-without-ed-step-loglikelihood-are-very-close." class="section level2">
<h2>Run mashr without ED step: loglikelihood are very close.</h2>
<div id="data-driven-covariance-1" class="section level3">
<h3>Data driven covariance</h3>
<pre class="r"><code>U.pca = cov_pca(data.strong,4)
U.c = cov_canonical(data.random)
m3 = mash(data.random, Ulist = c(U.pca,U.c))</code></pre>
<pre><code> - Computing 40000 x 393 likelihood matrix.
 - Likelihood calculations took 10.04 seconds.
 - Fitting model with 393 mixture components.
 - Model fitting took 86.23 seconds.
 - Computing posterior matrices.
 - Computation allocated took 155.79 seconds.</code></pre>
<pre class="r"><code>print(get_loglik(m))</code></pre>
<pre><code>[1] 205233</code></pre>
<pre class="r"><code>print(get_loglik(m3))</code></pre>
<pre><code>[1] 205232.7</code></pre>
<pre class="r"><code>barplot(get_estimated_pi(m3),las = 2)</code></pre>
<p><img src="figure/mashr_application.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.15

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mashr_0.2.21.0641 ashr_2.2-38      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2        plyr_1.8.4        compiler_3.5.1   
 [4] later_1.0.0       git2r_0.26.1      highr_0.7        
 [7] workflowr_1.5.0   iterators_1.0.12  tools_3.5.1      
[10] digest_0.6.18     evaluate_0.13     lattice_0.20-38  
[13] rlang_0.4.0       Matrix_1.2-15     foreach_1.4.7    
[16] yaml_2.2.0        parallel_3.5.1    mvtnorm_1.0-11   
[19] xfun_0.4          stringr_1.4.0     knitr_1.21       
[22] fs_1.3.1          rprojroot_1.3-2   grid_3.5.1       
[25] glue_1.3.0        R6_2.4.0          rmarkdown_1.11   
[28] mixsqp_0.1-97     rmeta_3.0         magrittr_1.5     
[31] whisker_0.3-2     backports_1.1.3   promises_1.1.0   
[34] codetools_0.2-16  htmltools_0.4.0   MASS_7.3-51.1    
[37] assertthat_0.2.1  abind_1.4-5       httpuv_1.5.2     
[40] stringi_1.3.1     doParallel_1.0.15 pscl_1.5.2       
[43] truncnorm_1.0-8   SQUAREM_2017.10-1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
