---
title: "mashr_application1"
author: "Yunqi Yang"
date: "12/8/2019"
output:
  pdf_document: default
  html_document: default
---

Remarks: 

1. Used data from "Shared genetic effects on chromatin and gene expression indicate a role for enhancer priming in immune response" https://doi.org/10.1038/s41588-018-0046-7. eQTLs from 4 conditions downloaded from https://zenodo.org/record/1158560#.Xgjl5BdKhp8.

2. For selecting strong signals: we first filter with the criteria p-value <0.05. Then, take the top signals (the most significant SNP for each gene) in naive condition, and filter gene-SNP pairs in other conditions based on top signals in naive condition. The resulting "top_snps.RData" contains 15678 significant Gene-SNP pairs in 4 conditions.

3. The 4 conditions are: naive/ IF/ IF+SL/ SL. (IF, SL are two drug treatments)

4. (1) Use strong signals to learn data-driven covariance patterns. (2) Fit models on the random subset of data, to learn weights for data-driven covariance matrix + canonical covariance matrix. 

5. Ways to assess the model fit. 

6. Compared the log-likelihood with/without ED step: extremely similar. 


```{r  }

library(ashr)
library(mashr)

```



## Format data to mashr format 
```{r  }

load("/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/top_snps.RData")
load("/Users/nicholeyang/Desktop/Rotation/mash-single-cell-rnaseq/data/random_snps.RData")

dt_beta_top <- top_snps[,c("beta_naive", "beta_IF", "beta_IFSL", "beta_SL")]
dt_pval_top <- top_snps[,c("p_nominal_naive", "p_nominal_IF", "p_nominal_IFSL", "p_nominal_SL")]


dt_beta_random <- random_snps[,c("beta_naive", "beta_IF", "beta_IFSL", "beta_SL")]
dt_pval_random <- random_snps[,c("p_nominal_naive", "p_nominal_IF", "p_nominal_IFSL", "p_nominal_SL")]

dt_beta_top = as.matrix(dt_beta_top)
dt_pval_top = as.matrix(dt_pval_top)

dt_beta_random = as.matrix(dt_beta_random)
dt_pval_random = as.matrix(dt_pval_random)

```



## Correlation structure
```{r  }

data.temp = mash_set_data(dt_beta_random, Shat = NULL, pval = dt_pval_random)
Vhat = estimate_null_correlation_simple(data.temp)
rm(data.temp)

```

```{r  }

data.strong = mash_set_data(dt_beta_top, Shat = NULL, pval = dt_pval_top, V=Vhat)
data.random = mash_set_data(dt_beta_random, Shat = NULL, pval = dt_pval_random, V=Vhat)

```


## Data driven covariance
```{r  }

U.pca = cov_pca(data.strong,4)
U.ed = cov_ed(data.strong, U.pca)

```



## Fit mashr model with both canonical & data-driven covariance structure
```{r  }

U.c = cov_canonical(data.random)
m = mash(data.random, Ulist = c(U.ed,U.c))

## Use learned weights to gain posterior estimation for strong signals
m2 = mash(data.strong, g=get_fitted_g(m), fixg=TRUE)

```





## Estimate data driven covariance
```{r  }

U.pca = cov_pca(data.strong,4)
U.ed = cov_ed(data.strong, U.pca)

```



## Fit mashr model with both canonical & data-driven covariance structure
```{r  }

U.c = cov_canonical(data.random)
m = mash(data.random, Ulist = c(U.ed,U.c))

## Use learned weights to gain posterior estimation for strong signals
m2 = mash(data.strong, g=get_fitted_g(m), fixg=TRUE)
```


```{r  }

print(get_loglik(m))
print(get_estimated_pi(m))
barplot(get_estimated_pi(m),las = 2)

```


```{r  }

mash_plot_meta(m,get_significant_results(m)[1])

```


## Effect Size Comparison 

### Posterior Mean VS. Original Effect

Conditions 1-4 are: naive/ IF/ IF+SL/ SL. (IF, SL are two drug treatments)

#### random signal posterior comparison
```{r  }

par(mfrow = c(2,2))

for (i in c(1:4)){
  
  plot(data.random$Bhat[,i], get_pm(m)[,i], pch = 20, ylab = "Posterior Mean", xlab = "Original Effect", main = paste('Condition_',i, sep = ""))
  abline(coef = c(0,1), col = "red")
  
}

```



#### Scaled by standard deviation: Posterior Mean VS. Original Effect 
```{r  }

par(mfrow = c(2,2))
for ( i in c(1:4)){
  
  plot(data.random$Bhat[,i]/data.random$Shat[,i], get_pm(m)[,i]/get_psd(m)[,i], pch = 20, ylab = "Posterior Mean", xlab = "Original Effect", main = paste('Condition_',i, sep = ""))
  abline(coef = c(0,1), col = "red")
  
}

```



#### strong signal posterior comparison
```{r  }

par(mfrow = c(2,2))

for (i in c(1:4)){
  
  plot(data.strong$Bhat[,i], get_pm(m2)[,i], pch = 20, ylab = "Posterior Mean", xlab = "Original Effect", main = paste('Condition_',i, sep = ""))
  abline(coef = c(0,1), col = "red")
  
}

```



#### Scaled by standard deviation: Posterior Mean VS. Original Effect 
```{r  }

par(mfrow = c(2,2))
for ( i in c(1:4)){
  
  plot(data.strong$Bhat[,i]/data.strong$Shat[,i], get_pm(m2)[,i]/get_psd(m2)[,i], pch = 20, ylab = "Posterior Mean", xlab = "Original Effect", main = paste('Condition_',i, sep = ""))
  abline(coef = c(0,1), col = "red")
  
}

```




## Run mashr without ED step: loglikelihood are very close. 

###  Data driven covariance
```{r  }

U.pca = cov_pca(data.strong,4)
U.c = cov_canonical(data.random)
m3 = mash(data.random, Ulist = c(U.pca,U.c))

```

```{r  }

print(get_loglik(m))
print(get_loglik(m3))
barplot(get_estimated_pi(m3),las = 2)


```




